core/src/com/unciv/models/ruleset/Building.kt

```diff
@@ -255,9 +256,45 @@ class Building : RulesetStatsObject(), INonPerpetualConstruction {
                 UniqueType.Unbuildable ->
                     yield(RejectionReasonType.Unbuildable.toInstance())
 
-                UniqueType.OnlyAvailable ->
-                    if (!unique.conditionalsApply(civ, cityConstructions.city))
+                UniqueType.OnlyAvailable -> {
+                    var displayable = false
+                    var shouldYield = false
+                    for (conditional in unique.conditionals) {
+                        if (Conditionals.conditionalApplies(unique, conditional, StateForConditionals(civ, cityConstructions.city)))
+                            continue
+                        when (conditional.type) {
+                            UniqueType.ConditionalBuildingBuiltAmount -> {
+                                val building = civ.getEquivalentBuilding(conditional.params[0]).name
+                                val amount = conditional.params[1].toInt()
+                                val cityFilter = conditional.params[2]
+                                val numberOfCities = civ.cities.count {
+                                    it.cityConstructions.containsBuildingOrEquivalent(building) && it.matchesFilter(cityFilter)
+                                }
+                                if (numberOfCities < amount)
+                                {
+                                    displayable = true
+                                    yield(RejectionReasonType.RequiresBuildingInSomeCities.toInstance(
+                                        "Requires a [$building] in at least [$amount] cities" +
+                                            " ($numberOfCities/$numberOfCities)"))
+                                }
+                            }
+                            UniqueType.ConditionalBuildingBuiltAll -> {
+                                val building = civ.getEquivalentBuilding(conditional.params[0]).name
+                                val cityFilter = conditional.params[1]
+                                if(civ.cities.any { it.matchesFilter(cityFilter)
+                                    !it.isPuppet && !it.cityConstructions.containsBuildingOrEquivalent(building)
+                                }) {
+                                    displayable = true
+                                    yield(RejectionReasonType.RequiresBuildingInAllCities.toInstance(
+                                        "Requires a [${building}] in all cities"))
+                                }
+                            }
+                            else -> { shouldYield = true}
+                        }
+                    }
+                    if (!displayable && shouldYield)
```

Comment:
Why do we need both of these? What are the two conditions we're trying to avoid? I only see one, if displayable is false then shouldYield will always be true, no?

Reply:
No, if all conditionals are valid, it shouldn't yield the rejection. It's ugly, yes, but not sure the better solution

Reply:
I just realized I'm dumb here. If we reach the else condition in here, we should just immediately yield and not need these bools at all